examples/pil-fun/src/Language/PilFun/Pretty.idr

```diff
@@ -19,72 +19,110 @@ import System.Random.Pure.StdGen
+  E : NameIsNew l [<] [<] Empty x
+  J : (0 _ : So $ x /= s) -> NameIsNew l funs vars ss x -> NameIsNew l funs vars (JustNew @{ss} s @{sub}) x
+  F : (0 _ : So $ x /= s) -> NameIsNew l funs vars ss x -> NameIsNew l (funs:<fun) vars (NewFun @{ss} {isInfix} {isPure} s @{sub} @{infixCond}) x
+  V : (0 _ : So $ x /= s) -> NameIsNew l funs vars ss x -> NameIsNew l funs ((:<) vars var mut) (NewVar @{ss} s @{sub}) x
 
 public export
 interface NamesRestrictions where
   reservedKeywords : SortedSet String
 
-rawNewName : Fuel -> (Fuel -> Gen MaybeEmpty String) =>
-             (vars : Vars) -> (funs : Funs) -> (names : UniqNames funs vars) ->
-             Gen MaybeEmpty (s ** NameIsNew names s)
+rawNewName : Fuel -> (l : SupportedLanguage) -> (Fuel -> Gen MaybeEmpty String) =>
+             (funs : Funs) -> (vars : Vars) -> (names : UniqNames l funs vars) ->
+             Gen MaybeEmpty (s ** NameIsNew l funs vars names s)
 
 export
-genNewName : Fuel -> (Fuel -> Gen MaybeEmpty String) =>
+genNewName : {l : SupportedLanguage} -> Fuel -> Gen MaybeEmpty String ->
              NamesRestrictions =>
-             (funs : Funs) -> (vars : Vars) -> (names : UniqNames funs vars) ->
-             Gen MaybeEmpty (s ** NameIsNew names s)
-genNewName fl @{genStr} funs vars names = do
-  nn@(nm ** _) <- rawNewName fl @{genStr} vars funs names
+             (funs : Funs) -> (vars : Vars) -> (names : UniqNames l funs vars) ->
+             Gen MaybeEmpty (s ** NameIsNew l funs vars names s)
+genNewName fl genStr funs vars names = do
+  nn@(nm ** _) <- rawNewName fl l @{const genStr} funs vars names
   if reservedKeywords `contains'` nm
-    then assert_total $ genNewName fl @{genStr} funs vars names -- we could reduce fuel instead of `assert_total`
+    then assert_total $ genNewName fl genStr funs vars names -- we could reduce fuel instead of `assert_total`
     else pure nn
 
-varName : UniqNames funs vars => IndexIn vars -> String
+varName : UniqNames l funs vars => IndexIn vars -> String
 varName @{JustNew @{ss} _} i         = varName @{ss} i
 varName @{NewFun @{ss} _}  i         = varName @{ss} i
 varName @{NewVar       s}  Here      = s
 varName @{NewVar @{ss} _}  (There i) = varName @{ss} i
 
-funName : UniqNames funs vars => IndexIn funs -> String
+funName : UniqNames l funs vars => IndexIn funs -> String
 funName @{JustNew @{ss} _} i         = funName @{ss} i
 funName @{NewFun       s}  Here      = s
 funName @{NewFun @{ss} _}  (There i) = funName @{ss} i
 funName @{NewVar @{ss} _}  i         = funName @{ss} i
 
-isFunInfix : UniqNames funs vars => IndexIn funs -> Bool
+isFunInfix : UniqNames l funs vars => IndexIn funs -> Bool
 isFunInfix @{JustNew @{ss} _} i        = isFunInfix @{ss} i
 isFunInfix @{NewFun {isInfix} _} Here  = isInfix
 isFunInfix @{NewFun @{ss} s} (There i) = isFunInfix @{ss} i
 isFunInfix @{NewVar @{ss} s} i         = isFunInfix @{ss} i
 
+isFunPure : UniqNames l funs vars -> IndexIn funs -> Bool
```

Comment:
Maybe, signatures of existing projection functions like `varName`, `funName` and `isFunInfix`  should be changed like in this function, i.e. passing `UniqNames` explicitly

Reply:
Fixed by 8c6e2aa4